Training/python/split_network_2p5.py

```diff
@@ -0,0 +1,196 @@
+  reduced_inputs = []
+  input_layers   = []
+  for comp_id in range(len(net_config.comp_names)):
+    comp_name = net_config.comp_names[comp_id]
+    n_comp_features = net_config.n_comp_branches[comp_id]
+    input_layer_comp = Input(name="input_{}_{}".format(loc, comp_name), shape=(1,1,n_comp_features))
+    input_layers.append(input_layer_comp)
+    input_layer_resh = keras.layers.Reshape(name='{}_reshape'.format(input_layer_comp.name), target_shape=(n_comp_features,))(input_layer_comp)
+    comp_net_setup.ComputeLayerSizes(n_comp_features)
+    # here we replace 2d with 1d (1x1 -> dense)
+    reduced_comp = reduce_n_features_1d(input_layer_resh, comp_net_setup, "{}_{}".format(loc, comp_name), net_config.first_layer_reg, basename='conv')
+    reduced_inputs.append(reduced_comp)
+
+  if len(net_config.comp_names) > 1:
+    # here we omit axis because now we have a dense layer -> shape is (None, N_feat_sum)
+    conv_all_start = Concatenate(name="{}_cell_concat".format(loc))(reduced_inputs)
+    comp_merge_net_setup.ComputeLayerSizes(conv_all_start.shape.as_list()[1]) # was 3 for conv, now is 1
+    # here we replace 2d with 1d (1x1 -> dense)
+    prev_layer = reduce_n_features_1d(conv_all_start, comp_merge_net_setup, "{}_all".format(loc), basename='conv')
+  else:
+    prev_layer = reduced_inputs[0]
+
+  return keras.Model(input_layers, prev_layer, name=loc)
+
+full_model = LoadModel(args.input)
+
+inner_model = ConvDenseAlias('inner')
+outer_model = ConvDenseAlias('outer')
+core_model  = CoreModel(inner=inner_model.output, outer=outer_model.output)
+
+copy_weights(model=full_model, target=inner_model)
+copy_weights(model=full_model, target=outer_model)
+copy_weights(model=full_model, target=core_model )
+
+assert test_prediction(model=inner_model, reference=full_model), "Error: '{}' model predictions differs from the full model ones".format(inner_model.name)
+assert test_prediction(model=outer_model, reference=full_model), "Error: '{}' model predictions differs from the full model ones".format(outer_model.name)
+assert test_prediction(model=core_model , reference=full_model), "Error: '{}' model predictions differs from the full model ones".format(core_model.name)
+
+with open(args.output+"/inner_summary.txt", "w") as smr:
+  inner_model.summary(print_fn=lambda x: smr.write(x+'\n'))
+with open(args.output+"/outer_summary.txt", "w") as smr:
+  outer_model.summary(print_fn=lambda x: smr.write(x+'\n'))
+with open(args.output+"/core_summary.txt", "w") as smr:
+  core_model.summary(print_fn=lambda x: smr.write(x+'\n'))
+with open(args.output+"/full_summary.txt", "w") as smr:
+  full_model.summary(print_fn=lambda x: smr.write(x+'\n'))
+
+for layer in core_model.inputs:
+  layer._name = 'input_inner' if 'inner' in layer.name else 'input_outer' if 'outer' in layer.name else layer.name 
+assert 'input_inner' in [l.name for l in core_model.inputs] and 'input_outer' in [l.name for l in core_model.inputs], "Core model input layer names are wrong"
+
```

Comment:
If this slash postfix matters, then it would be good also to rename the output layers (dropout) for inner/outer models to `{inner/outer}_all_dropout_4/Identity` (compatibility with [DeepTauId.cc#L1730](https://github.com/cms-tau-pog/cmssw/blob/5b56ea4f85b845ed3238c59310f79185ff481ab5/RecoTauTag/RecoTau/plugins/DeepTauId.cc#L1730))

Reply:
see previous two sections